src/lint/linter/UberCheckstyleLinter.php

```diff
@@ -5,270 +5,338 @@
  */
 class UberCheckstyleLinter extends ArcanistFutureLinter {
 
-  private $checkstyleScript = null;
-  private $checkstyleJar = null;
-  private $checkstyleConfig = null;
-  private $useScript = false;
-  private $maxFiles = 100;
-
-  public function getInfoName() {
-    return 'Java checkstyle linter';
-  }
-
-  public function getLinterName() {
-    return 'CHECKSTYLE';
-  }
-
-  public function getInfoURI() {
-    return 'http://checkstyle.sourceforge.net';
-  }
-
-  public function getInfoDescription() {
-    return 'Use checkstyle to perform static analysis on Java code';
-  }
-
-  public function getLinterConfigurationName() {
-    return 'checkstyle';
-  }
-
-  public function getLinterConfigurationOptions() {
-    $options = array(
-      'checkstyle.script' => array(
-        'type' => 'optional string',
-        'help' => pht('Checkstyle script to execute. Script must output checkstyle in XML to $stdout'),
-      ),
-      'checkstyle.jar' => array(
-        'type' => 'optional string',
-        'help' => pht('Checkstyle jar [from https://sourceforge.net/projects/checkstyle/files/checkstyle/]'),
-      ),
-      'checkstyle.config' => array(
-        'type' => 'optional string',
-        'help' => pht('Checkstyle configuration file'),
-      ),
-      'checkstyle.maxfiles' => array(
-        'type' => 'optional int',
-        'help' => pht('The maximum number of files to check per call of checkstyle. If there are more files, they will be split up into multiple checkstyle invocations.'),
-      ),
-    );
-    return $options + parent::getLinterConfigurationOptions();
-  }
-
-  public function setLinterConfigurationValue($key, $value) {
-    switch ($key) {
-      case 'checkstyle.script':
-        $this->checkstyleScript = $value;
-        return;
-      case 'checkstyle.jar':
-        $this->checkstyleJar = $value;
-        return;
-      case 'checkstyle.config':
-        $this->checkstyleConfig = $value;
-        return;
-      case 'checkstyle.maxfiles':
-        $this->maxFiles = $value;
-        return;
+    private $checkstyleScript = null;
+    private $checkstyleJar = null;
+    private $checkstyleConfig = null;
+    // default checkstyle URL to download the binary from if not specified in checkstyle.url
+    private $checkstyleURL = "https://sourceforge.net/projects/checkstyle/files/checkstyle/8.2/checkstyle-8.2-all.jar";
```

Comment:
There should be no default here as it will cause weird issues if the correct checkstyle version is not used consistently. The url should always be set by the consumers to specify which version of checkstyle they want to use.

Reply:
👍  Also, default will get dated.  It's better to force everyone to have to opt in to download.